src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -301,7 +301,8 @@
                          ;; we only sync columns that are either singlely indexed or is the first key in a composite index
                          (first (map :column_name (sort-by :ordinal_position idx-values)))))
           vals
-          set)))))
+          set
+          (disj nil))))))
```

Comment:
this is a safeguard for cases where the jdbc sometimes returns an index that has nil in column name.
I found this while working with sqlserver. will add an explanation

Reply:
Ah, I see this is dependent on line 301's `when-let `.

`(-> group-by update-vals vals set disj)` can be replaced by `(group-by vals keep set)` and I think this would be easier to understand.

Reply:
nice. did that in https://github.com/metabase/metabase/pull/36665/commits/0213cac7bd033eb6df64b2f3fb06ca145258c6fc